build/dsc/resource.rb

```diff
@@ -110,7 +110,12 @@ def ensure_value
           'present'
         elsif ensure_property.values.any? { |v| v.casecmp('enable') == 0 }
           'enable'
+        elsif ensure_property.values.any? { |v| v.casecmp('published') == 0 }
```

Comment:
Ok, so looking at https://github.com/PowerShell/xSCSMA/blob/dev/DSCResources/MSFT_xRunbookDirectory/MSFT_xRunbookDirectory.schema.mof#L4 I see the basis for this PR.

However, I think what you want to do here is treat `published` OR `draft` as a valid ensure value (at runtime).  However, the change here applies to the type generation (not looking at the resource at runtime) - so this only takes effect for the MOF schema parsing / type generation.

I think what you'll find, when you generate the types, is that you will only see `published` show up as an allowed `ensure` value and if you tried to use `draft`, it wouldn't do anyting.  You need to inform the Puppet side that either of these values are valid values for `ensure`, and that they should trigger enforcement behavior in Puppet.

I'm not sure how well that is going to work, but try it and find out once you're able to update this PR and include the generated types here.  I think it might be a little wonky since we've built our mapping layer around just the concepts of `ensure` / `absent`, but not any third state.